napari/_qt/widgets/qt_viewer_dock_widget.py

```diff
@@ -352,12 +350,6 @@ def __init__(
                 layout.addWidget(
                     self.close_button, 0, Qt.AlignmentFlag.AlignHCenter
                 )
-            layout.addWidget(
-                self.hide_button, 0, Qt.AlignmentFlag.AlignHCenter
-            )
-            layout.addWidget(
-                self.float_button, 0, Qt.AlignmentFlag.AlignHCenter
-            )
```

Comment:
I will also say that if we go ahead with the idea of allowing users to provide their own custom title bars, I think it's better if people define their own Qt custom title bar class in their own code, and pass that in to napari somehow.

Just deleting some bits out of the napari codebase to suit one specific use case is probably not going to work well as soon as the next group or several come along with their own conflicting ideas about how they want to customize title bars a slightly different way.